### ISSUE TITLE
Multiple Probing Bug

### ISSUE THREAD

**Issue by classicrocker883** [mriscoc/Ender3V2S1#319]
When selecting a different probe multiple, selecting 1 will automatically bring it to zero 0

When selecting 2, it probe 3 times.
and if 1 -> 0 is selected, it probes only once

The times it probes should be consistent with what value you select for how many probes you want.

**Comment by mriscoc** [comment]
Multiple probing means more than 1 probe, so, 1 is an invalid value and the correct behavior is disable it (0).
There is an additional probe because EXTRA_PROBING is enabled.

**Comment by classicrocker883** [comment]
it should let me probe the bed only 2 times because 2 gives me 3 probe times.
like u said it cannot be 0, so it just adds to the already probe amount of 1. and if you try to select 1 for 2 probes, it wont.

### this is how I fixed it

(line ~2290 in file proui/dwin.cpp)

```    void ApplyProbeMultiple() { PRO_data.multiple_probing = (MenuData.Value > 0) ? MenuData.Value : 0; }```

I changed the value to be greater than 0 rather than 1.

it should be called Extra Probing
because it adds to how many more probes on top of 1, which is basically mandatory.

so what it does is if in the menu you change it to 0, the bed probes 1 time.

if you change it to 1, it probes 2 times, 2 would be 3 times.

0 multiple probes means zero more added

**Comment by MarkforU** [comment]
The first touch, is for location, so setting for one probe gives you 2 touches. First one being location 2nd one being the variable that is put into the mesh. Now I have mine set to 3. So I get 4 probes per point, but it averages the 3 points for the best value from what I've read on Marlin. So from what I read, setting to 0 won't give you any mesh, but setting it to 1 will give you a value from the one probe spot, doing 2 will give not much of a average between 2 probe touches, but entering 3 will give you the average of 3 probe spots, but yes it will touch bed 4 times...I noticed this when doing HS probing and the first touch is very slow, then next touches are rapid. You can set number to 5 and watch as the probe reacts. Moves to spot, does a slow probe, then 5 fast probes after that. Then off to next spot. Now I haven't set it to 0 yet to confirm what Marlin says that mesh won't be built. Anyway, that was my take from it. I'm running silent 4.2.2 and 4.2.7 boards in each of my machines. I always set for 3 probes 8x8 because of the crazy jump in the z-axis when it gets to the middle far right side when printing...the head will go way up in the air.

**Comment by mriscoc** [comment]
![image](https://user-images.githubusercontent.com/2745567/186997211-834c7308-24df-4acb-b27d-4e633398c5a1.png)

**Comment by MarkforU** [comment]
Thank you, that clears it up!

**Comment by classicrocker883** [comment]
@MarkforU 
>I noticed this when doing HS probing and the first touch is very slow, then next touches are rapid. You can set number to 5 and watch as the probe reacts. Moves to spot, does a slow probe, then 5 fast probes after that. 

it probes slow first then fast? ive not seen it do this, just slow. i mean when first homing, it does a fast, then slow. but when actually probing its only slow.

so what would be more accurate....

probing many times in a row and getting the average.
or having it probe once or twice, but do it over again and see if that value changes?

maybe this is the tolerances but i get a change of +/- 0.01 anytime i probe the same spot.
so basically it will say like 0.03, then 0.04, back to 0.03, maybe go up to 0.05, then back to 0.04.

ive experimented with having it probe once or twice, to a few times and its the same.

so multiple probing really doesnt give u anything more accurate, anything more than 2 or 3  doesnt matter, because if u think of it, even if the value is different by 0.01 from each point, it wont affect the average very much when the tolerances are that low

also
@mriscoc 
when i changed the 1 to a 0 in here

` void ApplyProbeMultiple() { PRO_data.multiple_probing = (MenuData.Value > 0) ? MenuData.Value : 0; }`

then go into the menu and select 1 for mutliple probing, it probes twice right? but I get 1 fast, and 1 slow.

any other value and they are all slow probes.

id say this is a bug, whether or not it needs to be fixed well im not sure. i mean it would be nice to have 1 fast and 1 slow.
